### ISSUE TITLE
MapillaryOutsideGPXTrackError

### ISSUE THREAD

**Issue by xdTAGCLAN** [mapillary/mapillary_tools#611]
### Basic information
* Release version: Latest
* System: Windows
* Capture Device: Insta360 X3
* GPS Recorder: Android

### Steps to reproduce behavior

 1. Export 360 video to MP4
 2. Get .gpx file from phone
 3. Upload with this command via command prompt: 
```
mapillary_tools video_process C:\Users\......\Downloads\Mapillary\VID_20230403_184245_00_162(1).mp4 --geotag_source "gpx" --geotag_source_path C:\Users\.....\Downloads\Mapillary\VID_20230403_184245_00_162.gpx --video_sample_distance -1 --video_sample_interval 2
```
 

### Expected behavior
Successful upload
 
### Actual behavior
Error message: 
```
2023-04-09 17:24:35,581 - INFO    -      565 image(s) read in total
2023-04-09 17:24:35,581 - ERROR   -           565 image(s) failed due to MMJhwmfTYJVFm9TiUdmYYyrf6Pzpzf3YN1
2023-04-09 17:24:35,582 - ERROR   - MapillaryProcessError: Failed to process 565 files. To skip these errors, specify --skip_process_errors
```

### Additional information
Full CMD error and other info:
```
2023-04-09 17:18:30,344 - INFO    - Extracting video information: ffprobe -print_format json -hide_banner -show_format -show_streams C:\Users\.....\Downloads\Mapillary\VID_20230403_184245_00_162(1).mp4
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'C:\Users\.....\Downloads\Mapillary\VID_20230403_184245_00_162(1).mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    creation_time   : 2023-04-03T17:42:35.000000Z
    encoder         : Lavf58.13.101
  Duration: 00:18:50.04, start: 0.000000, bitrate: 200858 kb/s
  Stream #0:0[0x1](und): Video: h264 (Constrained Baseline) (avc1 / 0x31637661), yuvj420p(pc, bt709, progressive), 5760x2880, 200722 kb/s, 24 fps, 24 tbr, 24k tbn (default)
    Metadata:
      creation_time   : 2023-04-03T17:42:35.000000Z
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
    Side data:
      stereo3d: 2D
      spherical: equirectangular (0.000000/0.000000/0.000000)
  Stream #0:1[0x2](und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 129 kb/s (default)
    Metadata:
      creation_time   : 2023-04-03T17:42:35.000000Z
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
2023-04-09 17:18:30,514 - INFO    - Extracting frames: ffmpeg -hide_banner -nostdin -i C:\Users\.....\Downloads\Mapillary\VID_20230403_184245_00_162(1).mp4 -vf fps=1/2.0 -qscale:v 2 C:\Users\.....\Downloads\Mapillary\mapillary_sampled_video_frames\.mly_ffmpeg_VID_20230403_184245_00_162(1).mp4_11568_1681053510\VID_20230403_184245_00_162(1)_NA_%06d.jpg
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'C:\Users\.....\Downloads\Mapillary\VID_20230403_184245_00_162(1).mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    creation_time   : 2023-04-03T17:42:35.000000Z
    encoder         : Lavf58.13.101
  Duration: 00:18:50.04, start: 0.000000, bitrate: 200858 kb/s
  Stream #0:0[0x1](und): Video: h264 (Constrained Baseline) (avc1 / 0x31637661), yuvj420p(pc, bt709, progressive), 5760x2880, 200722 kb/s, 24 fps, 24 tbr, 24k tbn (default)
    Metadata:
      creation_time   : 2023-04-03T17:42:35.000000Z
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
    Side data:
      stereo3d: 2D
      spherical: equirectangular (0.000000/0.000000/0.000000)
  Stream #0:1[0x2](und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 129 kb/s (default)
    Metadata:
      creation_time   : 2023-04-03T17:42:35.000000Z
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> mjpeg (native))
Output #0, image2, to 'C:\Users\.....\Downloads\Mapillary\mapillary_sampled_video_frames\.mly_ffmpeg_VID_20230403_184245_00_162(1).mp4_11568_1681053510\VID_20230403_184245_00_162(1)_NA_%06d.jpg':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf59.27.100
  Stream #0:0(und): Video: mjpeg, yuvj420p(pc, bt709, progressive), 5760x2880, q=2-31, 200 kb/s, 0.50 fps, 0.50 tbn (default)
    Metadata:
      creation_time   : 2023-04-03T17:42:35.000000Z
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
      encoder         : Lavc59.37.100 mjpeg
    Side data:
      cpb: bitrate max/min/avg: 0/0/200000 buffer size: 0 vbv_delay: N/A
      stereo3d: 2D
      spherical: equirectangular (0.000000/0.000000/0.000000)
frame=  565 fps=1.6 q=2.0 Lsize=N/A time=00:18:50.00 bitrate=N/A speed=3.15x
video:619610kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
2023-04-09 17:24:32,216 - WARNING - Force the option "filetypes" to be "image" to avoid processing and uploading both the video samples and the videos themselves
Interpolating: 100%|¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦| 565/565 [00:02<00:00, 228.98images/s]
Processing: 100%|¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦| 565/565 [00:00<?, ?images/s]
Verifying image EXIF writing: 0images [00:00, ?images/s]
2023-04-09 17:24:35,580 - INFO    - Check the description file for details: C:\Users\.....\Downloads\Mapillary\mapillary_sampled_video_frames\mapillary_image_description.json
2023-04-09 17:24:35,581 - INFO    -      565 image(s) read in total
2023-04-09 17:24:35,581 - ERROR   -           565 image(s) failed due to MapillaryOutsideGPXTrackError
2023-04-09 17:24:35,582 - ERROR   - MapillaryProcessError: Failed to process 565 files. To skip these errors, specify --skip_process_errors
```

**Comment by xdTAGCLAN** [comment]
> Likely to be the GPX and video clocks are not synced. See https://github.com/mapillary/mapillary_tools#geotagging-with-gpx
> 
> You can probably solve it by adding `--interpolation_use_gpx_start_time` but before that, double check the following:
> 
> * the `mapillary_image_description.json` file
> * first timestamp of the GPX,
> * and the video creation time and duration with
> 
> ```
> ffprobe -print_format json -hide_banner -show_format -show_streams C:\Users\.....\Downloads\Mapillary\VID_20230403_184245_00_162(1).mp4
> ```
> 
> Make sure the video creation time - duration is falling into the GPX timestamp range.

Hm, appears the time is both the same. Still isn't working....

**Comment by ptpt** [comment]
I see. It should be the bug that has been fixed in https://github.com/mapillary/mapillary_tools/pull/602

The fix is included in the latest release https://github.com/mapillary/mapillary_tools/releases/tag/v0.10.1

To upgrade:
```
pip install -U mapillary_tools
```

**Comment by xdTAGCLAN** [comment]
> I see. It should be the bug that has been fixed in #602
> 
> The fix is included in the latest release https://github.com/mapillary/mapillary_tools/releases/tag/v0.10.1
> 
> To upgrade:
> 
> ```
> pip install -U mapillary_tools
> ```

It doesn't seem to be fixed.. But what did work was looking in mapillary_image_description.json and looking at how many seconds the first image was off by from the gpx, which is stated in the first message: "The image date time is 1143.042 seconds behind the GPX start point".

So I added --interpolation_use_gpx_start_time 8343 (7200 + 1143, because the .gpx was not correct with when the video was taken). Which still caused a few of the images to be discarded: 
2023-04-13 17:27:59,305 - INFO    -      565 image(s) read in total
2023-04-13 17:27:59,305 - INFO    -           518 image(s) are ready to be uploaded
2023-04-13 17:27:59,305 - WARNING -             1 image(s) skipped due to MapillaryOutsideGPXTrackError
2023-04-13 17:27:59,306 - WARNING -            46 image(s) skipped due to MapillaryDuplicationError

To upload the files I replaced "video_process" with "video_process_and_upload" which successfully worked! :
Uploading IMAGE (1/2): 100%|████████████████████████████████████████████████████████| 533M/533M [01:29<00:00, 6.21MB/s]
Uploading IMAGE (2/2): 100%|██████████████████████████████████████████████████████| 21.0M/21.0M [00:04<00:00, 5.43MB/s]
2023-04-13 17:29:49,443 - INFO    -        2  IMAGE sequences uploaded
2023-04-13 17:29:49,443 - INFO    -    554.2M data in total
2023-04-13 17:29:49,444 - INFO    -    554.2M data uploaded
2023-04-13 17:29:49,444 - INFO    -     94.1s upload time

Should I worry about some of the images still not working?

**Comment by xdTAGCLAN** [comment]
It appears that it has worked!
https://www.mapillary.com/app/user/xdTAGCLAN?lat=52.18699079999999&lng=5.368134800000007&z=17&pKey=260469102986818&focus=photo&x=0.2743018357292758&y=0.6104614589860088&zoom=0
